docs/source/basic/operators.rst

```diff
@@ -0,0 +1,126 @@
+Operators
+=========
+
+The operator describes the operation applied to each element of a range of input elements. There are two ways to define an ``operator``: a lambda or a functor.
```

Comment:
What about `std::function`, function pointers, ...?

Reply:
I didn't tested it. The main problem is CUDA. Some templates are not correctly resolved if there is a dependency between host code and a `__device__` function. That's the reason, why global `__device__` does not work. But to be hones, I don't know if functions pointers and `std::function` are working or not.

Reply:
`std::function` will not work with CUDA: https://nvidia.github.io/libcudacxx/standard_api/utility_library/functional.html

Reply:
Okay, maybe also explain this here.

Reply:
I extend the warning box with a explanation.